ext/src/Cassandra/DefaultTable.c

```diff
@@ -0,0 +1,411 @@
+  RETURN_ZVAL(value, 0, 1);
+}
+
+PHP_METHOD(DefaultTable, compactionStrategyClassName)
+{
+  cassandra_table* self;
+  zval* value;
+
+  if (zend_parse_parameters_none() == FAILURE)
+    return;
+
+  self = (cassandra_table*) zend_object_store_get_object(getThis() TSRMLS_CC);
+
+  php_cassandra_get_schema_field(self->meta, "compaction_strategy_class", &value TSRMLS_CC);
+  RETURN_ZVAL(value, 0, 1);
+}
+
+PHP_METHOD(DefaultTable, compactionStrategyOptions)
+{
+  cassandra_table* self;
+  zval* value;
+
+  if (zend_parse_parameters_none() == FAILURE)
+    return;
+
+  self = (cassandra_table*) zend_object_store_get_object(getThis() TSRMLS_CC);
+
+  php_cassandra_get_schema_field(self->meta, "compaction_strategy_options", &value TSRMLS_CC);
+  RETURN_ZVAL(value, 0, 1);
+}
+
+PHP_METHOD(DefaultTable, compressionParameters)
+{
+  cassandra_table* self;
+  zval* value;
+
+  if (zend_parse_parameters_none() == FAILURE)
+    return;
+
+  self = (cassandra_table*) zend_object_store_get_object(getThis() TSRMLS_CC);
+
+  php_cassandra_get_schema_field(self->meta, "compression_parameters", &value TSRMLS_CC);
+  RETURN_ZVAL(value, 0, 1);
+}
+
+static zval*
+php_cassandra_create_column(cassandra_ref*        schema,
+                            const CassSchemaMeta* metadata TSRMLS_DC)
+{
+  zval*             zcolumn;
+  cassandra_column* column;
+
+  const CassSchemaMetaField* field;
+  const CassValue*           value;
+
+  const char* validator;
+  size_t      validator_length;
+
+  field = cass_schema_meta_get_field(metadata, "validator");
+  value = cass_schema_meta_field_value(field);
+
+  ASSERT_SUCCESS_VALUE(cass_value_get_string(value, &validator,
+                                             &validator_length), NULL);
+
+  MAKE_STD_ZVAL(zcolumn);
+  object_init_ex(zcolumn, cassandra_default_column_ce);
+
+  column = (cassandra_column*) zend_object_store_get_object(zcolumn TSRMLS_CC);
+
+  ASSERT_SUCCESS_BLOCK(php_cassandra_get_schema_field(metadata, "column_name",
+                                                      &column->name TSRMLS_CC),
+    zval_ptr_dtor(&zcolumn);
+    return NULL;
+  );
+
+  if (php_cassandra_parse_column_type(validator, validator_length,
+                                      &column->reversed, &column->frozen,
+                                      &column->type TSRMLS_CC) == FAILURE) {
+    zval_ptr_dtor(&zcolumn);
+    return NULL;
+  }
+
+  column->meta   = metadata;
+  column->schema = php_cassandra_add_ref(schema);
+
+  return zcolumn;
+}
+
+PHP_METHOD(DefaultTable, column)
+{
+  cassandra_table* self;
+  char* name;
+  int   name_len;
+  zval* column;
+  const CassSchemaMeta* metadata;
+
+  if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "s", &name, &name_len) == FAILURE) {
+    return;
+  }
+
+  self     = (cassandra_table*) zend_object_store_get_object(getThis() TSRMLS_CC);
+  metadata = cass_schema_meta_get_entry(self->meta, name);
+
+  if (metadata == NULL) {
+    RETURN_NULL();
```

Comment:
nitpick: Inconsistency for NULL value return; unless RETURN_NULL() is doing something different than return null.